_rules/SVG-image-has-name-7d6734.md

```diff
@@ -0,0 +1,205 @@
+- [ARIA10: Using aria-labelledby to provide a text alternative for non-text content](https://www.w3.org/WAI/WCAG21/Techniques/aria/ARIA10)
+
+## Test Cases
+
+### Passed
+
+#### Passed example 1
+
+The `<svg>` element has an implicit role of `graphics-document`, has visible content rendered by the `<circle>` element and, is included in the accessibility tree by the `<title>` does not contain only whitespace.
+
+```html
+<svg xmlns="http://www.w3.org/2000/svg" width="100" height="100">
+  <title>A yellow circle</title>
+	<circle cx="50" cy="50" r="40" fill="yellow"></circle>
+</svg>
+```
+
+#### Passed example 2
+
+The `<svg>` element has an explicit role of `img`, has visible content rendered by the `<circle>` element, and is included in the accessibility tree by the `<title>` that does not contain only whitespace.
+
+```html
+<svg xmlns="http://www.w3.org/2000/svg" role="img" width="100" height="100">
+  <title>A yellow circle</title>
+	<circle cx="50" cy="50" r="40" fill="yellow"></circle>
+</svg>
+```
+
+#### Passed example 3
+
+The `<svg>` element has an explicit role of `img`, has visible content rendered by the `<circle>` element, and is included in the accessibility tree by the `aria-label` attribute that is not only whitespace.
+
+```html
+<svg xmlns="https://www.w3.org/2000/svg" role="img" width="100" height="100" aria-label="A yellow circle">
+	<circle cx="50" cy="50" r="40" fill="yellow"></circle>
+</svg>
+```
+
+#### Passed example 4
+
+The `<svg>` child `<g>` element has visible content, is included in the accessibility tree with the `graphics-object` role, and provides an accessible name through the `aria-label` attribute that is not only whitespace.
+
+```html
+<svg xmlns="https://www.w3.org/2000/svg" width="600" height="400" viewBox="200 0 400 400" role="presentation">
+	<g role="graphics-object" aria-label="small yellow rectangle in larger green rectangle" transform="translate(30,-90)"> 
+		<rect fill="green" width="50" height="90"/> 
+		<rect fill="yellow" x="5" y="5" width="40" height="30" />
+	</g>
+</svg>
+```
+
+### Failed
+
+#### Failed example 1
+
+The `<svg>` element is included in the accessibility tree with an explicit role of `img`, has visible content rendered by the `<circle>` element, but has no accessible name.
+
+```html
+<svg xmlns="http://www.w3.org/2000/svg" role="img" width="100" height="100">
+	<circle cx="50" cy="50" r="40" stroke="green" stroke-width="4" fill="yellow"></circle>
+</svg>
+```
+
+#### Failed example 2
+
+The `<svg>` element is included in the accessibility tree with an explicit role of `img`, has visible content rendered by the `<circle>` element, but has only has whitespace in the `<title>` element that is being used for the accessible name.
+
+```html
+<svg xmlns="http://www.w3.org/2000/svg" role="img" width="100" height="100" >
+	<title> </title>
+  	<circle role="none" cx="50" cy="50" r="40" fill="yellow"></circle>
```

Comment:
Is it relevant for this example to have `role="none"`? I can't find a reason.

Reply:
Thanks Carlos.